File: VB/ResistantPopup/Popup/RepositoryItemCustomPopupContainerEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator

Namespace ResistantPopup.Popup
	<UserRepositoryItem("RegisterCustomEditor")> _
	Public Class RepositoryItemCustomPopupContainerEdit
		Inherits RepositoryItemPopupContainerEdit
		Shared Sub New()
			RegisterCustomEdtor()
		End Sub

		Friend Const CustomName As String = "CustomPopupContainerEdit"
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return CustomName
			End Get
		End Property

		Public Shared Sub RegisterCustomEdtor()
			EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(CustomName, GetType(CustomPopupContainerEdit), GetType(RepositoryItemCustomPopupContainerEdit), GetType(PopupContainerEditViewInfo), New ButtonEditPainter(), True))
		End Sub
	End Class
End Namespace

File: CS/ResistantPopup/Popup/RepositoryItemCustomPopupContainerEdit.cs
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;

namespace ResistantPopup.Popup {
    [UserRepositoryItem("RegisterCustomEditor")]
    public class RepositoryItemCustomPopupContainerEdit :RepositoryItemPopupContainerEdit {
        static RepositoryItemCustomPopupContainerEdit() {
            RegisterCustomEdtor();
        }

        internal const string CustomName = "CustomPopupContainerEdit";
        public override string EditorTypeName {
            get { return CustomName; }
        }

        public static void RegisterCustomEdtor() {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(CustomName,
                typeof(CustomPopupContainerEdit), typeof(RepositoryItemCustomPopupContainerEdit),
                typeof(PopupContainerEditViewInfo), new ButtonEditPainter(), true));
        }
    }
}
